### PR TITLE
Add provisioning key expiry date to generateDeviceProvisioningKey

### PR DESCRIPTION
**Pull request by nitishagar**
Change-Type: patch
Signed-off-by: Nitish Agarwal <1592163+nitishagar@users.noreply.github.com>

See: https://jel.ly.fish/product-improvement-provisioning-api-key-management-4c86a2c3-8899-4764-9019-8c9b932e57f5

### CODE REVIEW

File: lib/models/application.ts [hunk 1]
```diff
@@ -988,6 +994,7 @@ const getApplicationModel = function (
 			slugOrUuidOrId: string | number,
 			keyName?: string,
 			keyDescription?: string,
+			keyExpiryDate?: string,
```
Comment (thgreasi) [inline lib/models/application.ts:997]: I think a Date object might work as well, can you give it a try?

File: lib/models/application.ts [hunk 2]
```diff
@@ -979,6 +980,11 @@ const getApplicationModel = function (
 		 * });
 		 *
 		 * @example
+		 * balena.models.application.generateProvisioningKey(123, '<name>', '<description>', '<YYYY-MM-DD>').then(function(key) {
```
Comment (thgreasi) [inline lib/models/application.ts:983]: Let's give 
```suggestion
		 * balena.models.application.generateProvisioningKey(123, 'api key name', 'api key long description', '2022-01-01').then(function(key) {
```

**Review by thgreasi — COMMENTED** [review]
I test case would be nice as well.
Feel free to self-certify after my two comments.

File: lib/models/application.ts [hunk 3]
```diff
@@ -1001,6 +1008,7 @@ const getApplicationModel = function (
 						roles: ['provisioning-api-key'],
 						name: keyName,
 						description: keyDescription,
+						expiry_date: keyExpiryDate,
```
Comment (thgreasi) [inline lib/models/application.ts:1011]: Did you try this?
Shouldn't it be v ?
```suggestion
						expiryDate: keyExpiryDate,
```
See: https://github.com/balena-io/open-balena-api/blob/37dee1df959c196af3e5c5d8cda4da8cf8d4acce/src/features/api-keys/lib.ts#L104

File: tests/integration/models/application.spec.ts [hunk 4]
```diff
@@ -776,10 +776,11 @@ describe('Application Model', function () {
 			});
 
 			describe('balena.models.application.generateProvisioningKey()', function () {
-				const getProvisioningKeys = async function (appNameOrSlug) {
+				const getProvisioningKeys = async function (appNameOrSlug, options?) {
```
Comment (thgreasi) [inline tests/integration/models/application.spec.ts:779]:
```suggestion
				const getProvisioningKeys = async function (appNameOrSlug, options) {
```

**Comment by nitishagar** [comment]
@balena-ci retest

**Comment by nitishagar** [comment]
@balena-ci retest

**Comment by thgreasi** [comment]
@nitishagar `@balena-ci retest` doesn't work here, since appveyor is not part of balena-ci.
You need to open appveyor, login & issue a restart from there. Let me do that for you,

**Comment by nitishagar** [comment]
> @nitishagar `@balena-ci retest` doesn't work here, since appveyor is not part of balena-ci. You need to open appveyor, login & issue a restart from there. Let me do that for you,

Thanks @thgreasi. I didn't see any `re-run` option on `appveyor` maybe permission issue - checking with @ops.

From the re-run the `TEST_ONLY_ON_ENVIRONMENT=browser` one passed but the `node` one failed. Seems intermittent again; but will confirm once permission issue sorted.

**Review by thgreasi — APPROVED** [review]

**Pull request merged**